src/main/java/com/xebialabs/maven/mustache/AbstractMustachifierMojo.java

```diff
@@ -69,6 +82,39 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
             flush(fileset.getDirectory());
         }
+        for (SingleFile sFile: files) {
+            saveSingleFile(sFile);
+        }
+    }
+
+    final void saveSingleFile(final SingleFile aFile) {
+        TFile input = new TFile(aFile.getSourceFile());
+        TFile output = new TFile(aFile.getTargetFileName());
+        try {
+            String content = TFiles.toString(input, Charset.forName(aFile.getModelEncoding()));
+            if (aFile.isFilterOnlyMustacheProperties()) {
+                content = filterMustacheOnly(content);
+            }
+            final String processed = transform(content);
+            File parent = new File(output.getParent());
+            parent.mkdirs();
+            save(processed, output, aFile.getModelEncoding());
+        } catch (MojoExecutionException e) {
+            getLog().error("Failed to read input file: "+input, e);
+        } catch (IOException e) {
+            getLog().warn("Failed to read input file: "+input);
+        }
+    }
+
+    final String filterMustacheOnly(final String aContent) {
```

Comment:
I provide a bit  of explanation into javadoc of SingleFile#setFilterOnlyMustacheProperties